Automattic/wp-calypso
# Add breadcrumb info into state.

### Changes proposed in this Pull Request

* Add breadcrumb info to global state
* Use the global state to retrieve the breadcrumb items
* Use state actions to update breadcrumb on Plugins Browser and Plugin Details page

### Testing instructions
**Go back to search results** (#58501)
* Go to `/plugins` page
* Search for a plugin. Ex: `yoast`
* Select a plugin from this search
* You should be able to see the `Search Results` on the breadcrumb, as below
* Click on `Search Results` and you should be redirected to the previous search results page
<img width="342" alt="Screen Shot 2022-01-26 at 16 34 18" src="https://user-images.githubusercontent.com/5039531/151394139-e1307a5e-cc7a-4270-b74c-67458659f5df.png">

**Opening the Plugin Details directly**
* Delete the indexed DB entirely or just remove the breadcrumb piece
* Access a plugin page directly. Ex: `/plugins/woocommerce-bookings`
* You should see `Plugins > {Plugin Name}` as breadcrumb
<img width="342" alt="Screen Shot 2022-01-26 at 16 34 29" src="https://user-images.githubusercontent.com/5039531/151395128-0400853c-0d90-492f-8592-a669c2d26d43.png">

---

Fixes  #58501
